packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkNativeInitialization.kt

```diff
@@ -0,0 +1,337 @@
+        return when (trackingConsent?.lowercase(Locale.US)) {
+            "pending" -> TrackingConsent.PENDING
+            "granted" -> TrackingConsent.GRANTED
+            "not_granted" -> TrackingConsent.NOT_GRANTED
+            else -> {
+                Log.w(
+                    DdSdk::class.java.canonicalName,
+                    "Unknown consent given: $trackingConsent, " +
+                            "using ${TrackingConsent.PENDING} as default"
+                )
+                TrackingConsent.PENDING
+            }
+        }
+    }
+
+    private fun buildSite(site: String?): DatadogSite {
+        val siteLower = site?.lowercase(Locale.US)
+        return when (siteLower) {
+            "us1", "us" -> DatadogSite.US1
+            "eu1", "eu" -> DatadogSite.EU1
+            "us3" -> DatadogSite.US3
+            "us5" -> DatadogSite.US5
+            "us1_fed", "gov" -> DatadogSite.US1_FED
+            "ap1" -> DatadogSite.AP1
+            else -> DatadogSite.US1
+        }
+    }
+
+    private fun buildVitalUpdateFrequency(vitalsUpdateFrequency: String?): VitalsUpdateFrequency {
+        val vitalUpdateFrequencyLower = vitalsUpdateFrequency?.lowercase(Locale.US)
+        return when (vitalUpdateFrequencyLower) {
+            "never" -> VitalsUpdateFrequency.NEVER
+            "rare" -> VitalsUpdateFrequency.RARE
+            "average" -> VitalsUpdateFrequency.AVERAGE
+            "frequent" -> VitalsUpdateFrequency.FREQUENT
+            else -> VitalsUpdateFrequency.AVERAGE
+        }
+    }
+
+    private fun buildUploadFrequency(uploadFrequency: String?): UploadFrequency {
+        return when (uploadFrequency?.lowercase(Locale.US)) {
+            "rare" -> UploadFrequency.RARE
+            "average" -> UploadFrequency.AVERAGE
+            "frequent" -> UploadFrequency.FREQUENT
+            else -> UploadFrequency.AVERAGE
+        }
+    }
+
+    private fun buildBatchSize(batchSize: String?): BatchSize {
+        return when (batchSize?.lowercase(Locale.US)) {
+            "small" -> BatchSize.SMALL
+            "medium" -> BatchSize.MEDIUM
+            "large" -> BatchSize.LARGE
+            else -> BatchSize.MEDIUM
+        }
+    }
+
+    internal fun getConfigurationFromJSONFile(): DdSdkConfiguration {
+        val jsonString = jsonFileReader.parseAssetsJSONFile(appContext, "datadog-configuration.json")
+
+        val configuration = Gson().fromJson(jsonString, JSONConfigurationFile::class.java).configuration
+
+        return configuration.asDdSdkConfiguration()
+    }
+
+    companion object {
+        /**
+         * Initializes the Datadog React Native SDK from your MainApplication.
+         *
+         * @param appContext: The application context of your React Native application.
+         */
+        @JvmStatic
+        @Suppress("TooGenericExceptionCaught")
+        fun initFromNative(appContext: Context) {
+            val nativeInitialization = DdSdkNativeInitialization(appContext)
```

Comment:
you may want to call explicitly `appContext.applicationContext`, because user may pass UI-based context instead.